snuba/query/snql/parser.py

```diff
@@ -35,17 +72,28 @@
     visit_parameters_list,
     visit_quoted_literal,
 )
+from snuba.util import parse_datetime
+
 
 snql_grammar = Grammar(
     r"""
-    query_exp             = match_clause where_clause? collect_clause? group_by_clause? having_clause? order_by_clause?
+    query_exp             = match_clause select_clause group_by_clause? where_clause? having_clause? order_by_clause? limit_by_clause? limit_clause? offset_clause? sample_clause?
```

Comment:
Does this mean we are finally allowing a SAMPLE cllause applied to the full query?
If yes I thinnk we need to define the semantics of a sampled joined query when the sampling key is different between the two (or more) entities that are being joined and it is not the same as the join key.

Example:
```
events sample key is event_id
groups sample key is group_id
join key is group_id
```
Clickhouse will want us to provide the sampling clause on one of the joined table or on both, the result will be different between sampling events or sampling the groups.